pycqed/measurement/measurement_control.py

```diff
@@ -615,44 +632,251 @@ def set_sweep_points_2D(self, sweep_points_2D):
     the 2D plotmon (which does a heatmap) and the adaptive plotmon.
     '''
 
+    def _get_plotmon_axes_info(self):
```

Comment:
I couldn't follow everything that is done in this function, but in broad strokes it looks fine and, again, it has been working well at XLD
Maybe one thing that could help make this function more readable is to add short statements before the chunks of code that do something specific, like "Now we deal with compression by updating update param_x". A good example what you wrote at line 807. For instance I'm not entire sure what is done in the large if statement starting below line 716.